crates/sui-core/src/authority_active/checkpoint_driver/mod.rs

```diff
@@ -274,18 +274,13 @@ pub async fn checkpoint_process<A>(
 
         // (3) Create a new proposal locally. This will also allow other validators
         // to query the proposal.
-        // Only move to propose when we have the full checkpoint certificate
-        let sequence_number = state_checkpoints.lock().next_checkpoint();
-        if sequence_number > 0 {
-            // Check that we have the full certificate for the previous checkpoint.
-            // If not, we are not ready yet to make a proposal.
-            if !matches!(
-                state_checkpoints.lock().get_checkpoint(sequence_number - 1),
-                Ok(Some(AuthenticatedCheckpoint::Certified(..)))
-            ) {
-                tokio::time::sleep(timing.delay_on_local_failure).await;
-                continue;
-            }
+        // If we have already signed a new checkpoint locally, there is nothing to do.
+        if matches!(
+            state_checkpoints.lock().latest_stored_checkpoint(),
+            Some(AuthenticatedCheckpoint::Signed(..))
```

Comment:
Here we wait on the checkpoint being signed, whereas before we waited on the checkpoint not being authenticated. I am not certain of the reason / difference here?

Reply:
Here we wait on the certified checkpoint. If it's signed, we wait and go back the the beginning of the loop